tests/prod_preactivation_test.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python3
+
+from testUtils import Utils
+from Cluster import Cluster
+from WalletMgr import WalletMgr
+from Node import Node
+from Node import ReturnType
+from TestHelper import TestHelper
+
+import decimal
+import re
+import time
+
+###############################################################
+# nodeos_run_test
+# --dump-error-details <Upon error print etc/eosio/node_*/config.ini and var/lib/node_*/stderr.log to stdout>
+# --keep-logs <Don't delete var/lib/node_* folders upon test completion>
+###############################################################
+
+Print=Utils.Print
+errorExit=Utils.errorExit
+cmdError=Utils.cmdError
+from core_symbol import CORE_SYMBOL
+
+args = TestHelper.parse_args({"--host","--port","--prod-count","--defproducera_prvt_key","--defproducerb_prvt_key","--mongodb"
+                              ,"--dump-error-details","--dont-launch","--keep-logs","-v","--leave-running","--only-bios","--clean-run"
+                              ,"--sanity-test","--p2p-plugin","--wallet-port"})
+server=args.host
+port=args.port
+debug=args.v
+enableMongo=args.mongodb
+defproduceraPrvtKey=args.defproducera_prvt_key
+defproducerbPrvtKey=args.defproducerb_prvt_key
+dumpErrorDetails=args.dump_error_details
+keepLogs=args.keep_logs
+dontLaunch=args.dont_launch
+dontKill=args.leave_running
+prodCount=2 # args.prod_count
+onlyBios=args.only_bios
+killAll=args.clean_run
+sanityTest=args.sanity_test
+p2pPlugin=args.p2p_plugin
+walletPort=args.wallet_port
+
+Utils.Debug=debug
+localTest=True
+cluster=Cluster(host=server, port=port, walletd=True, enableMongo=enableMongo, defproduceraPrvtKey=defproduceraPrvtKey, defproducerbPrvtKey=defproducerbPrvtKey)
+walletMgr=WalletMgr(True, port=walletPort)
+testSuccessful=False
+killEosInstances=not dontKill
+killWallet=not dontKill
+dontBootstrap=sanityTest
+
+WalletdName=Utils.EosWalletName
+ClientName="cleos"
+timeout = .5 * 12 * 2 + 60 # time for finalization with 1 producer + 60 seconds padding
+Utils.setIrreversibleTimeout(timeout)
+
+try:
+    TestHelper.printSystemInfo("BEGIN prod_preactivation_test.py")
+    cluster.setWalletMgr(walletMgr)
+    Print("SERVER: %s" % (server))
+    Print("PORT: %d" % (port))
+
+    if enableMongo and not cluster.isMongodDbRunning():
+        errorExit("MongoDb doesn't seem to be running.")
+
+    if localTest and not dontLaunch:
+        cluster.killall(allInstances=killAll)
+        cluster.cleanup()
+        Print("Stand up cluster")
+        if cluster.launch(pnodes=prodCount, totalNodes=prodCount, prodCount=prodCount, onlyBios=onlyBios, dontBootstrap=dontBootstrap, p2pPlugin=p2pPlugin, extraNodeosArgs=" --plugin eosio::producer_api_plugin") is False:
```

Comment:
Comments sounds like this test only expects one producer, but you allow prodCount to be set on the command line, and use it to set the number of producer nodes? Is that what you want to do?